From 42837b8134844c1d08014e480f9497d165c57ef6 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 23 Dec 2019 16:31:48 +0100
Subject: [PATCH] core: don't check error parameter of get_name_owner_handler()

It's a *return* parameter, not an input parameter. Yes, this is a bit
confusing for method call replies, but we try to use the same message
handler for all incoming messages, hence the parameter. We are supposed
to write any error into it we encounter, if we want, and our caller will
log it, but that's it.
---
 src/core/unit.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/core/unit.c b/src/core/unit.c
index 03b4b57..c54abe9 100644
--- a/src/core/unit.c
+++ b/src/core/unit.c
@@ -3218,11 +3218,6 @@ static int get_name_owner_handler(sd_bus_message *message, void *userdata, sd_bu
 
         u->get_name_owner_slot = sd_bus_slot_unref(u->get_name_owner_slot);
 
-        if (sd_bus_error_is_set(error)) {
-                log_error("Failed to get name owner from bus: %s", error->message);
-                return 0;
-        }
-
         e = sd_bus_message_get_error(message);
         if (sd_bus_error_has_name(e, "org.freedesktop.DBus.Error.NameHasNoOwner"))
                 return 0;
-- 
1.8.3.1

